### DISCUSSION TITLE
Getting a read-only node to run without a master

### DISCUSSION THREAD

**Discussion by computinglife** [rqlite/rqlite#1821] [General]
So we have been testing the tool. Its quite nifty and ran without any issues. 

One hiccup though that we had in our setup was that when the master is down, we couldnt bring up the read-only node. 

Anyway to do make this happen, so that the read-only node can continue serving the data at "none" (ofcourse no writes) consistency? 

When master comes up, it should be able to rejoin the cluster automatically, in the background and start recieving the updates like it normally does 

Is there a way to do this with any of the current options ?

**Comment by otoolep** [comment]
This is a limitation in the Raft library from Hashicorp. Consul has the same issues I believe. It’s not specific to rqlite. 
I think it’s fixable but I have not done the work to do so yet. 


Sent from Yahoo Mail for iPhone


On Wednesday, June 26, 2024, 9:57 AM, computinglife ***@***.***> wrote:




So we have been testing the tool. Its quite nifty and ran without any issues.

One hiccup though that we had in our setup was that when the master is down, we couldnt bring up the read-only node.

Anyway to do make this happen, so that the read-only node can continue serving the data at "none" (ofcourse no writes) consistency?

When master comes up, it should be able to rejoin the cluster automatically, in the background and start recieving the updates like it normally does

Is there a way to do this with any of the current options ?

**Comment by computinglife** [comment]
This a blocker for us to adopt rqlite. 

I wish to use it thought because it would be simpler than the alternative.

If its a quick fix, we could hold out though  -  please do let us know if there is a way out. 

I would have volunteered to help, but we don't have many with expertise in Go. 

Thanks

**Comment by otoolep** [comment]
I plan to work on shipping 9.0 before making any changes related to this issue. 9.0 is in active development, but it's weeks away from being released (perhaps 2-3 months). Changes I'm planning for 9.0 may make this easier to fix.

**Comment by computinglife** [comment]
Thank you. Hope you are able to fix this behavior. We will come back to re-eval this for our next product change. 
I will reach out to find if there are anyways to contribute.
